### DISCUSSION TITLE
Fzflua files NOT show all the files

### DISCUSSION THREAD

**Discussion by bo-wu** [ibhagwan/fzf-lua#1519] [General]
I have more than 400 files in my directory. WIth Telescope, it shows all the files in the list. However, with Fzflua files, it only shows 200 files. Where is the problem?

![image](https://github.com/user-attachments/assets/3bd311aa-3b54-4951-94ed-5983b1c42845)

**Comment by ibhagwan** [comment]
Use `:FzfLua files debug=true` and explore the command in the debug line, run it in the shell, whatever the output is will be displayed in fzf-lua.

Compare that with your Telescope fd options, there might be a difference due to the flag used (hidden, gitignore, etc).

**Reply by bo-wu** [reply]
> Use `:FzfLua files debug=true` and explore the command in the debug line, run it in the shell, whatever the output is will be displayed in fzf-lua.
> 

I did what you said, and found that  the output in terminal is more than that in FzfLua files.  So what's wrong?

@ibhagwan

**Reply by bo-wu** [reply]
Under the same directory,
In Neovim with fzf-lua
![image](https://github.com/user-attachments/assets/dabb2b94-e174-4801-a740-5c9393d9aecb)

In terminal
![image](https://github.com/user-attachments/assets/9969a8bf-6685-4331-ab6f-a3a2fac05ddf)


BTW, sometimes when the first time I open FzfLua files, it shows more than 300 files.

**Reply by ibhagwan** [reply]
> BTW, sometimes when the first time I open FzfLua files, it shows more than 300 files.

Perhaps a bug (unlikely but maybe?) or something in your config, can you try with mini.sh (top of the README)?

Perhaps also a bug relating to icons, try disabling git/file icons?

**Reply by bo-wu** [reply]
> Perhaps also a bug relating to icons, try disabling git/file icons?

You are right. After disable file icons, it show all the files.

**Reply by ibhagwan** [reply]
> > Perhaps also a bug relating to icons, try disabling git/file icons?
> 
> You are right. After disable file icons, it show all the files.

Can you run `:FzfLua files multiprocess=false` and see if you get an error?

**Reply by bo-wu** [reply]
> > > Perhaps also a bug relating to icons, try disabling git/file icons?
> > 
> > 
> > You are right. After disable file icons, it show all the files.
> 
> Can you run `:FzfLua files multiprocess=false` and see if you get an error?

I tried, both with file icons enable  and disable, there is no error.

**Reply by ibhagwan** [reply]
> > Can you run `:FzfLua files multiprocess=false` and see if you get an error?
> 
> I tried, both with file icons enable and disable, there is no error.

Can you try with minimal config? Do you see all files?
```lua
sh -c "$(curl -s https://raw.githubusercontent.com/ibhagwan/fzf-lua/main/scripts/mini.sh)"
```

also, if you have mini installed can you try `:FzfLua files file_icons=mini`?

**Reply by bo-wu** [reply]
> > > > > Perhaps also a bug relating to icons, try disabling git/file icons?
> > > > 
> > > > 
> > > > You are right. After disable file icons, it show all the files.
> > > 
> > > 
> > > Can you run `:FzfLua files multiprocess=false` and see if you get an error?
> > 
> > 
> > I tried, both with file icons enable and disable, there is no error.
> 
> Can you try with minimal config? Do you see all files?
> 
> ```lua
> sh -c "$(curl -s https://raw.githubusercontent.com/ibhagwan/fzf-lua/main/scripts/mini.sh)"
> ```

I tried this one, and Yes, it shows all files and with icons. So that means conflicts with other plugins or config?

**Reply by ibhagwan** [reply]
> > 
> > Can you try with minimal config? Do you see all files?
> > ```lua
> > sh -c "$(curl -s https://raw.githubusercontent.com/ibhagwan/fzf-lua/main/scripts/mini.sh)"
> > ```
> 
> I tried this one, and Yes, it shows all files and with icons. So that means conflicts with other plugins or config?

Probably something messed up with your nvim-web-devicons setup, I would still classify this behavior as a bug in fzf-lua as no matter what it shouldn’t crash while enumerating files.

- If you have mini installed can you try `:FzfLua files file_icons=mini`?
- Have you set any customizations for nvim-web-devicons?
- Do you have a link to your config files?

Solving this would require a bit of debugging, trying to disable other config options/plugins to see where the issue lies.

**Reply by bo-wu** [reply]
> > > Can you try with minimal config? Do you see all files?
> > > ```lua
> > > sh -c "$(curl -s https://raw.githubusercontent.com/ibhagwan/fzf-lua/main/scripts/mini.sh)"
> > > ```
> > 
> > 
> > I tried this one, and Yes, it shows all files and with icons. So that means conflicts with other plugins or config?
> 
> Probably something messed up with your nvim-web-devicons setup, I would still classify this behavior as a bug in fzf-lua as no matter what it shouldn’t crash while enumerating files.
> 
> * If you have mini installed can you try `:FzfLua files file_icons=mini`?

Yes, It's OK, but file icons don't show up any more. 

> * Have you set any customizations for nvim-web-devicons?

Not by myself. I use the default config in LazyVim. 

> * Do you have a link to your config files?

I searched the config added by myself, nothing to do with nvim-web-devicons

> 
> Solving this would require a bit of debugging, trying to disable other config options/plugins to see where the issue lies.

**Reply by ibhagwan** [reply]
> Not by myself. I use the default config in LazyVim.

Would you be able to open a bug issue and fill as much details as you can about your setup?
Which neovim version, fzf, shell, OS, fd/ripgrep versions, etc - so I can attempt to replicate this issue?

**Reply by ibhagwan** [reply]
I also have another suspect area, if I recall correctly Lazy uses a formatter by default, can you try with `:FzfLua files formatter=false`?

**Reply by bo-wu** [reply]
> > Not by myself. I use the default config in LazyVim.
> 
> Would you be able to open a bug issue and fill as much details as you can about your setup? Which neovim version, fzf, shell, OS, fd/ripgrep versions, etc - so I can attempt to replicate this issue?

OK. I will do it latter.

**Reply by bo-wu** [reply]
> I also have another suspect area, if I recall correctly Lazy uses a formatter by default, can you try with `:FzfLua files formatter=false`?

I tried, but with file_icons=true by default, it still shows part files as before.

**Reply by ibhagwan** [reply]
> > I also have another suspect area, if I recall correctly Lazy uses a formatter by default, can you try with `:FzfLua files formatter=false`?
> 
> I tried, but with file_icons=true by default, it still shows part files as before.

Then I guess it’s the icons bit, I’m just confused why the error doesn’t show up with:
```lua
:FzfLua files multiprocess=false
```
Can you run that again and check if there is a hidden exception  in `:messages`?

**Reply by bo-wu** [reply]
> > > I also have another suspect area, if I recall correctly Lazy uses a formatter by default, can you try with `:FzfLua files formatter=false`?
> > 
> > 
> > I tried, but with file_icons=true by default, it still shows part files as before.
> 
> Then I guess it’s the icons bit, I’m just confused why the error doesn’t show up with:
> 
> ```lua
> :FzfLua files multiprocess=false
> ```
> 
> Can you run that again and check if there is a hidden exception in `:messages`?

I did as you said, run command :messages after :FzfLua files multiprocess=false, still nothing show up.

**Reply by ibhagwan** [reply]
> I did as you said, run command :messages after :FzfLua files multiprocess=false, still nothing show up.

This bug obviously rare and somehow tied to your config so I’ll need all the information I can get in order to reproduce it, a link to your config and perhaps the project you’re working on (if not private) would be great to try and reproduce this bug.

**Reply by bo-wu** [reply]
> > I did as you said, run command :messages after :FzfLua files multiprocess=false, still nothing show up.
> 
> This bug obviously rare and somehow tied to your config so I’ll need all the information I can get in order to reproduce it, a link to your config and perhaps the project you’re working on (if not private) would be great to try and reproduce this bug.

I just opened an issue. The project is private, but I will try some open project and share the link here

**Reply by bo-wu** [reply]
It seems the same for any project that has more the 200 or 300 files. 

You can try this  https://github.com/SFML/SFML.git
